lockfile/src/parsers/yarn.rs

```diff
@@ -39,24 +40,20 @@ fn parse_entry(input: &str) -> Result<&str, Package> {
     })
 }
 
-fn entry_name(input: &str) -> Result<&str, &str> {
+fn entry_name(input: &str) -> IResult<&str, &str> {
     let (i, _) = opt(tag(r#"""#))(input)?;
     let opt_at = opt(tag("@"));
     let name = tuple((opt_at, take_until("@")));
     context("name", recognize(name))(i)
 }
 
-fn entry_version(input: &str) -> Result<&str, &str> {
+fn entry_version(input: &str) -> IResult<&str, &str> {
     let (i, _) = take_until(r#"version"#)(input)?;
     let version_key = tuple((tag(r#"version"#), opt(tag(r#"""#)), tag(r#" ""#)));
     context("version", delimited(version_key, is_version, tag(r#"""#)))(i)
 }
 
-fn is_version<T, E: nom::error::ParseError<T>>(input: T) -> IResult<T, T, E>
-where
-    T: nom::InputTakeAtPosition,
-    <T as nom::InputTakeAtPosition>::Item: AsChar,
-{
+fn is_version(input: &str) -> IResult<&str, &str> {
```

Comment:
This was the only place that we were using `nom::IResult` directly instead of using our alias. But it wasn't necessary, so I changed it to be like the rest of our code and use the alias.

Why was it written that way if it wasn't necessary? My bet is that the person who wrote this function just copied [`alphanumeric1`](https://docs.rs/nom/7.1.3/src/nom/character/complete.rs.html#583-589) and tweaked the body to fit our use case.